x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/original_scenarios.md

```diff
@@ -0,0 +1,264 @@
+```Gherkin
+Given user already has 8.x prebuilt rules installed in Kibana
+And there are new updates available for those prebuilt rules
+And user is on Rules Management Page
+When user navigates to the Rules Update tab
+Then user should see all the prebuilt rules that have updates available
+And user can update outdated prebuilt rules
+And successfully updated message is displayed
+And Rules Upgrade tab counter is decreased according to the number of updated rules
+```
+
+### Package Installation / Rule Installation / Rule Update Failure
+
+#### **Scenario: Error message is displayed when any prebuilt rules operation fails**
+
+```Gherkin
+Given user is <action> prebuilt rules
+When the installation or update process fails
+Then user should see an error message
+And prebuilt rules are not installed/updated
+And the callout message for pending installs/updates is still displayed on Rules Management Page
+And the number of available rules to install and upgrade in the badges does not change
+
+Examples:
+  | action                |
+  | installing all        |
+  | installing selected   |
+  | installing individual |
+  | updating all          |
+  | updating selected     |
+  | updating individual   |
+```
+
+#### **Scenario: No callout messages are displayed when rule package installation fails and no rules are avialble for install/update**
+
+```Gherkin
+Given user navigates to Rules Management Page
+And user is running a fresh instance
+And rule package installation fails
+Then no callouts message should be displayed
+And the number of available rules to install and upgrade in the badges does not change
+```
+
+### Add Elastic Rules Page
+
+#### **Scenario: New workflow elements are displayed on Rules Management Page**
+
+```Gherkin
+Given a user that doesn’t have `security_detection_engine` package installed
+When user is on Rules Management Page
+Then "+Add Elastic rules" menu with available Rules counter is displayed
+And Rule Updates tab is displayed
+And "+Add Elastic rules" button is displayed on empty Rules Table
+```
+
+#### **Scenario: Rules settings persist on Add Elastic Rules table**
+
+```Gherkin
+Given a user has Rules listed on Add Elastic Rules page
+When <case>
+Then the available rules state should persist for all the rules
+
+Examples:
+  | case                              |
+  | user reloads the page             |
+  | after switching table pagination  |
+  | after filtering and clear filters |
+```
+
+#### **Scenario: User can navigate back to Rules Management page**
+
+```Gherkin
+Given a user is on Add Rules Page
+When user navigates back to Rules Management page
+Then Rules Management Page is properly displayed
+```
+
+#### **Scenario: User can filter prebuilt rules by rule name or by tag**
+
+```Gherkin
+Given a user is on Add Rules Page
+When user filters by <filter>
+Then Add Rules Table is properly updated
+
+Examples:
+  | filter                  |
+  | rule name on search bar |
+  | Tag filter              |
+```
```

Comment:
I think this one is too vague so I removed it.

- We should add detailed scenarios to the `Rule installation workflow: filtering, sorting, pagination` section.